components/RenderActivityLog.js

```diff
@@ -0,0 +1,95 @@
+//import liraries
+import React from "react";
+import { View, Text, Pressable } from "react-native";
+import { useNavigation } from "@react-navigation/native";
+
+import { ScreenConstants } from "constants";
+
+import PrefetchCacheRecord from "components/PrefetchCacheRecord";
+import { ChevronDownIcon, ChevronUpIcon } from "components/Icon";
+import { localStyles } from "./RenderActivityLog.styles";
+import useStyles from "hooks/use-styles";
+import useType from "hooks/use-type";
+
+// create a component
+const RenderActivityLog = ({
+  logs,
+  index,
+  accordionState,
+  handleAccordionChange,
+}) => {
+  const { styles, globalStyles } = useStyles(localStyles);
+  const { getTabScreenFromType } = useType();
+  const navigation = useNavigation();
+
+  if (!logs || accordionState.length === 0) return null;
+
+  return (
+    <>
+      <View
+        style={{
+          ...globalStyles.rowContainer,
+          justifyContent: "space-between",
+          marginVertical: 5,
+        }}
+      >
+        <Pressable
+          onPress={() => {
+            const type = logs[1][0]?.post_type;
```

Comment:
Will this crash the app if `logs` only has a single 0th entry ?

Reply:
Hi @zdmc23. The logs are being grouped in such a way that logs[0] will be the name of the group & logs[1] will be an array of the activity logs related to that group. It will work this way even if there is only one activity in the log. If there are no activities then null is returned and the Card is not rendered. I have attached some screenshots for your reference. 

<img width="1074" alt="Screenshot 2022-06-27 at 1 23 41 PM" src="https://user-images.githubusercontent.com/104883753/175888708-918bafa1-72a0-4c66-b6db-1bc60921d4dc.png">

<img width="1073" alt="Screenshot 2022-06-27 at 1 09 54 PM" src="https://user-images.githubusercontent.com/104883753/175887293-10545b25-3f6a-4b7e-b4a8-556329ddd6ec.png">

<img width="1077" alt="Screenshot 2022-06-27 at 1 13 05 PM" src="https://user-images.githubusercontent.com/104883753/175887312-f5fd8926-caf5-46fa-940d-0382c8659e74.png">